### PR TITLE
Add element InsertAdjacentElement method

### PR DESCRIPTION
**Pull request by miketonks**
https://github.com/dennwc/dom/issues/29

### CODE REVIEW

File: element.go [hunk 1]
```diff
@@ -274,3 +274,8 @@ func (e *Element) AttachShadow(opts AttachShadowOpts) *ShadowRoot {
 	m["delegatesFocus"] = opts.DeligatesFocus
 	return AsShadowRoot(e.v.Call("attachShadow", js.ValueOf(m)))
 }
+
+// InsertAdjacentElement inserts a given element node at a given position relative to the element it is invoked upon.
+func (e *Element) InsertAdjacentElement(position string, newElement *Element) js.Value {
```
Comment (dennwc) [inline element.go]: I think it's worth making `position` a separate enum type with all 4 supported values.

Comment (miketonks) [reply element.go]: Are you suggesting just a type, like this:

```
type Color string
```

or actual validation, perhaps like this?

https://blog.learngoprogramming.com/golang-const-type-enums-iota-bc4befd096d3

Comment (dennwc) [reply element.go]: Yes, custom string type with a set of constants will be enough.

Comment (miketonks) [reply element.go]: updated

**Pull request merged**